dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,526 @@
+            )
+        }
+    }
+
+    // Get the publicly available data for a CollectionGroup by id
+    //
+    pub fun getCollectionGroupData(id: UInt64): DSSCollection.CollectionGroupData {
+        pre {
+            DSSCollection.collectionGroupByID[id] != nil: "Cannot borrow collection group, no such id"
+        }
+
+        return DSSCollection.CollectionGroupData(id: id)
+    }
+
+    // Get the publicly available data for a CollectionGroup by id
+    //
+    pub fun validateTimeRange(timeBound: Bool, startTime: UFix64?, endTime: UFix64?): Bool {
+        if !timeBound {
+            return true
+        }
+
+        if startTime! <= getCurrentBlock().timestamp && endTime! >= getCurrentBlock().timestamp {
+            return true
+        } else {
+            return false
+        }
+    }
+
+    // A DSSCollection NFT
+    //
+    pub resource NFT: NonFungibleToken.INFT, MetadataViews.Resolver {
+        pub let id: UInt64
+        pub let collectionGroupID: UInt64
+        pub let serialNumber: UInt64
+        pub let completionDate: UFix64
+        pub let completedBy: String
+        pub let level: UInt64
+
+        pub fun name(): String {
+            let collectionGroupData: DSSCollection.CollectionGroupData
+                = DSSCollection.getCollectionGroupData(id: self.collectionGroupID)
+            let level: String = self.level.toString()
+            return collectionGroupData.name
+                .concat(" Level ")
+                .concat(level)
+                .concat(" Completion Token")
+        }
+
+        pub fun description(): String {
+            let serialNumber: String = self.serialNumber.toString()
+            let completionDate: String = self.completionDate.toString()
+            return "Completed by "
+                .concat(self.completedBy)
+                .concat(" on ")
+                .concat(completionDate)
+                .concat(" with serial number ")
+                .concat(serialNumber)
+        }
+
+        destroy() {
+            emit DSSCollectionNFTBurned(id: self.id)
+        }
+
+        pub fun getViews(): [Type] {
+            return [
+                Type<MetadataViews.Display>()
+            ]
+        }
+
+        pub fun resolveView(_ view: Type): AnyStruct? {
+            return MetadataViews.Display(
+                name: self.name(),
+                description: self.description(),
+                thumbnail: MetadataViews.HTTPFile(
+                    url:"https://storage.googleapis.com/dl-nfl-assets-prod/static/images/collection-group/token-placeholder.png"
+                )
+            )
+        }
+
+        init(
+            id: UInt64,
+            collectionGroupID: UInt64,
+            serialNumber: UInt64,
+            completedBy: String,
+            level: UInt64
+        ) {
+            pre {
+                DSSCollection.collectionGroupByID[collectionGroupID] != nil: "no such collectionGroupID"
+            }
+
+            self.id = id
```

Comment:
from my above comment, we can set the id to be the resource uuid `self.uuid`

Reply:
Done

Reply:
this still uses `self.id=id`

Reply:
Fixed it. Sorry I missed that.